tests/v2/validation/rbac/configmaps/configmaps_rbac_test.go

```diff
@@ -0,0 +1,276 @@
+				require.True(cm.T(), k8sError.IsForbidden(err))
+			}
+		})
+	}
+}
+
+func (cm *ConfigmapsRBACTestSuite) TestCreateConfigmapAsEnvVar() {
+	subSession := cm.session.NewSession()
+	defer subSession.Cleanup()
+
+	tests := []struct {
+		role   rbac.Role
+		member string
+	}{
+		{rbac.ClusterOwner, rbac.StandardUser.String()},
+		{rbac.ClusterMember, rbac.StandardUser.String()},
+		{rbac.ProjectOwner, rbac.StandardUser.String()},
+		{rbac.ProjectMember, rbac.StandardUser.String()},
+		{rbac.ReadOnly, rbac.StandardUser.String()},
+	}
+	for _, tt := range tests {
+		cm.Run("Validate config map creation for user with role "+tt.role.String(), func() {
+			adminProject, namespace, err := createProjectAndNamespace(cm.client, cm.cluster.ID)
+			require.NoError(cm.T(), err)
+
+			newUser, standardUserClient, err := rbac.CreateUserAndAddToClusterRole(cm.client, tt.member, tt.role.String(), cm.cluster, adminProject)
+			require.NoError(cm.T(), err)
+			cm.T().Logf("Created user: %v", newUser.Username)
+
+			downstreamWranglerContext, err := standardUserClient.WranglerContext.DownStreamClusterWranglerContext(cm.cluster.ID)
+			require.NoError(cm.T(), err)
+
+			configMapCreatedByUser, err := createConfigmap(namespace, downstreamWranglerContext)
+			switch tt.role.String() {
+			case rbac.ClusterOwner.String(), rbac.ProjectOwner.String(), rbac.ProjectMember.String():
+				require.NoError(cm.T(), err)
+				_, err = dep.CreateDeployment(standardUserClient, cm.cluster.ID, deploymentName, namespace.Name, newPodTemplate(configMapCreatedByUser.Name, ""), 1)
+				require.NoError(cm.T(), err)
+			case rbac.ClusterMember.String(), rbac.ReadOnly.String():
+				require.Error(cm.T(), err)
+				require.True(cm.T(), k8sError.IsForbidden(err))
+			}
+		})
+	}
+}
+
+func (cm *ConfigmapsRBACTestSuite) TestUpdateConfigmap() {
+	subSession := cm.session.NewSession()
+	defer subSession.Cleanup()
+
+	tests := []struct {
+		role   rbac.Role
+		member string
+	}{
+		{rbac.ClusterOwner, rbac.StandardUser.String()},
+		{rbac.ClusterMember, rbac.StandardUser.String()},
+		{rbac.ProjectOwner, rbac.StandardUser.String()},
+		{rbac.ProjectMember, rbac.StandardUser.String()},
+		{rbac.ReadOnly, rbac.StandardUser.String()},
+	}
+
+	for _, tt := range tests {
+		cm.Run("Validate config map creation for user with role "+tt.role.String(), func() {
+			adminProject, namespace, err := createProjectAndNamespace(cm.client, cm.cluster.ID)
+			require.NoError(cm.T(), err)
+
+			newUser, standardUserClient, err := rbac.CreateUserAndAddToClusterRole(cm.client, tt.member, tt.role.String(), cm.cluster, adminProject)
+			require.NoError(cm.T(), err)
+			cm.T().Logf("Created user: %v", newUser.Username)
+
+			downstreamWranglerContext, err := cm.client.WranglerContext.DownStreamClusterWranglerContext(cm.cluster.ID)
+			require.NoError(cm.T(), err)
+
+			_, err = createConfigmap(namespace, downstreamWranglerContext)
+			require.NoError(cm.T(), err)
+
+			configMap.Data["foo1"] = "bar1"
```

Comment:
where is the configMap object coming from? Did you miss assigning the created ConfigMap ( in line 156) to the configMap variable.